crucible-mir/src/Mir/Intrinsics.hs

```diff
@@ -1997,6 +2094,59 @@ mirVector_resizeIO bak _tpr mirVec newLenSym = do
     pure $ MirVector_PartialVector $ V.generate (fromInteger newLen) getter
 
 
+mirAggregate_uninitIO ::
+    IsSymBackend sym bak =>
+    bak ->
+    Word ->
+    IO (RegValue sym MirAggregateType)
+mirAggregate_uninitIO _bak sz = return $ MirAggregate sz mempty
+
+mirAggregate_uninitSymIO ::
+    IsSymBackend sym bak =>
+    bak ->
+    RegValue sym UsizeType ->
+    IO (RegValue sym MirAggregateType)
+mirAggregate_uninitSymIO bak szSym = do
+  sz <- case asBV szSym of
+    Just x -> return (BV.asUnsigned x)
+    Nothing -> addFailedAssertion bak $ Unsupported callStack $
+      "Attempted to allocate aggregate of symbolic size"
```

Comment:
This size-is-concrete check is repeated three different times throughout `Mir.Intrinsics`, so it would be worth factoring this part out into a helper function.